### ISSUE TITLE
Unable to display the .tif files exactly as displayed in the research paper

### ISSUE THREAD

**Issue by ShivamSRS** [MLNeurosurg/opensrh#2]
Hello 

I am opening this issue, because I am unable to display the .tif files included in the dataset exactly as they are displayed in the paper. In the paper, the images are displayed in pinkish-purple colour (see Figure 1 on page 3 in the research paper https://arxiv.org/pdf/2206.08439.pdf ); and my lab says those are how exactly the H&E stains must look like. 

![Alternate image text](https://i.imgur.com/xpCNhzu.png)

To try displaying this image, I first used the dataset code provided by the authors in the link https://github.com/MLNeurosurg/opensrh/tree/main/opensrh/datasets by using their SRHClassificationDataset class and passing only the get_srh_base_aug() for transforms. However, the images that I got to display looked nothing like the paper's. 

Since, the authors used the tifffile package in their github code a lot. I wrote this small code to copy only their importing the image, processing it as mentioned in the code and paper and displaying it

```
import tifffile
import numpy as np
image = tifffile.imread('dataset/studies/NIO_001/6/patches/NIO_001-6-2000_3000_300_600.tif').astype(np.float32)
print(image.shape,type(image))
# print(image_label_dictionary[4][0].shape)
tifffile.imshow(image[[0,1],:,:])
```

![Alternate image text](https://i.imgur.com/xiYAnAe.png)

This is different colour than expected. Then I perform the post processing required to add a third channel and form a new 3 channel image using bumpy
```
ch2 = image[0, :, :]
ch3 = image[1, :, :]
ch1 = ch3 - ch2 + 5000 / 65536.0
new_im = np.stack((ch1, ch2, ch3), axis=0)
tifffile.imshow(new_im)
```
![Alternate image text](https://i.imgur.com/K22qPyH.png)

This is displayed in again a different colour than what's expected.

Further I normalise the image to have all values between 0 and 255

```
norm_im = []

for i in (0,1,2):
#     print(new_im[[i],:,:].shape)
    norm_im.append(scaler.fit_transform(new_im[[i],:,:].reshape(300,300)))

norm_im = np.array(norm_im)
print(norm_im,norm_im.shape,type(norm_im),norm_im.max(),norm_im.min ())
tifffile.imshow(norm_im)
```

![Alternate image text](https://i.imgur.com/hIhEXDs.png)

which also gives a different than expected colour. 

I want to know why can I not display the H&E stains as displayed in the paper using the provided code and data. Do I not need that colour?  Please clarify on this, as my experiment results on this dataset can be much more significant , if the images I am displaying are similar to the images that are being displayed by other authors and researchers working in this area.

**Comment by chengjia63** [comment]
The virtual H&E (the blue-pink color schemes) shown in figure 1 in the paper is generated by the SRH microscope using proprietary software from the manufacturer, and are for physician visualization purposes. If your research requires H&E pseudocolored images, the wholeslide virtual H&E colored images are released as their original dicom format in OpenSRH. 

For our benchmarks and experiments, we generate separate RGB images using the strips that contain raw laser signals and transforms returned by `get_srh_base_aug()` as described in appendix A, and avoid the pseudocolored images because:

- The proprietary color scheme is designed to simulate H&E for physicians, but does not reflect any biological process;
- The coloring algorithm can change and contain improvements to the newer slides that introduce temporal inconsistencies; and most importantly
- The 8-bit virtual H&E stain coloring removes high frequency laser signals contained in the 16 bit output.

To visualize SRH images similar to appendix A, you can use the following code snippet:

```
import os
from typing import List

import numpy as np
import matplotlib.pyplot as plt

import torch
from torchvision.transforms import Compose
from opensrh.datasets.improc import process_read_im, get_srh_base_aug

def viz_rescale_hist(im: torch.Tensor, bits: List[float] = [7, 8, 8]) -> np.ndarray:
    """Rescale SRH images and convert to numpy for visualization."""

    def hist_eq(im, bits=8):
        """Rescale one channel of image.

        see https://ianwitham.wordpress.com/tag/histogram-equalization/
        """
        num_bins = np.floor(2**bits).astype(np.uint16)
        hist, bins = np.histogram(im.flatten(), num_bins)
        hist[0] = 0

        cdf = hist.cumsum()
        cdf = (num_bins - 1) * cdf / cdf[-1]  # normalize

        im2 = np.interp(im.flatten(), bins[:-1], cdf)
        return np.array(im2).reshape(im.shape)

    stacked = np.dstack([
        hist_eq(im[0, ...], bits[0]),
        hist_eq(im[1, ...], bits[1]),
        hist_eq(im[2, ...], bits[2]),
    ])
    return stacked.astype(np.uint8)


db_root = "/path/to/opensrh/"
im_path = os.path.join(db_root, "studies/NIO_001/6/patches/NIO_001-6-2000_3000_300_600.tif")
xf = Compose(get_srh_base_aug())
im = xf(process_read_im(im_path))

fig, ax = plt.subplots(1, 1)
ax.imshow(viz_rescale_hist(im)); ax.axis("off")
```

![NIO_001-6-2000_3000_300_600 tif](https://user-images.githubusercontent.com/16587525/221319959-3de82668-3f00-45a8-b43d-0d5d72e52aa4.png)
